### ISSUE TITLE
XXE vulnerability in GenericPackager

### ISSUE THREAD

**Issue by k-sever** [jpos/jPOS#462]
GenericPackager (as well as GenericValidatingPackager) are vulnerable to [XML External Entity (XEE) attack](https://owasp.org/www-community/vulnerabilities/XML_External_Entity_(XXE)_Processing).

Here is the test example that reproduces it 
```java
@Test
    public void testXXEAttach() {
        try {
            String xeeAttackXml = "<?xml version=\"1.0\" encoding=\"ISO-8859-1\"?>\n"
                + "<!DOCTYPE foo [\n"
                + "  <!ELEMENT foo ANY >\n"
                + "  <!ENTITY xxe SYSTEM \"https://jposxee.free.beeceptor.com/hacked\" >]>\n"
                + "<foo>&xxe;</foo>";
            new GenericPackager().readFile(new ByteArrayInputStream(xeeAttackXml.getBytes()));
            fail("Expected ISOException to be thrown");
        } catch (ISOException ex) {
            assertEquals("Entity extension is disabled", ex.getMessage(), "ex.getMessage()");
        }
    }
```
The `xxe` entity is getting resolved and request goes to `https://jposxee.free.beeceptor.com/hacked` (it's just a mock API):
<img width="1216" alt="Screenshot 2022-01-05 at 22 34 45" src="https://user-images.githubusercontent.com/8113027/148285444-1b422da0-a887-41d2-abc5-4c59c8b67e2c.png">

It's not a finalized test that you can check in, but it gives you the idea.

**Comment by ar** [comment]
GenericPackager configuration is usually read from the classpath. Having the ability to define entities is a nice to have feature that enhances the code. Somebody with enough access to the classpath to use an XXE have easier ways to compromise the system. 

Can you provide a use case where a GenericPackager configuration could be injected from a remote host?

**Comment by k-sever** [comment]
> GenericPackager configuration is usually read from the classpath.
Well, you can't enforce it. IMO it should be secure by default, security > code readability. I'd rather disable external entities and make it possible to configure XML reader features for GenericPackager, so that anyone who actually needs it can redefine it.

**Comment by ar** [comment]
Fair enough. I'll reopen the issue and think about a way to configure it to reenable the feature.
